### DISCUSSION TITLE
Feature request: sync a fork (upstream to origin) directly from GitHub web site

### DISCUSSION THREAD

**Discussion by r-pankevicius** [community/community#21628] [New to GitHub]
<p>That’s a very neat feature: Create a pull request from my fork to upstream directly from GitHub web site.</p>
<p>But reverse - sync a fork (upstream to origin) requires my to go to command line and do “git magic” there.</p>
<p>What about if there is a button to do pull from upstream to origin directly on web site?</p>
<p>I’d like to teach a girl how to do translations (localizations) for the open source project, but I see that she will require some help from me.</p>
<p>This could be made simpler: she herself creates a fork, does translations directly on GitHub web, commits there, makes a pull request and… here the magic ends: she can’t easily pull from upstream to master even if both resides in the same place.</p>
<p>I mean with such feature she wouldn’t need to have anything on her hard drive to be actively involved in project development.</p>

**Comment by canuckjacq** [comment]
<p>:wave: Welcome!</p>
<p>This is a popular feature request. I’ve added yours to the issue!</p>
<p>In the meantime, I had a search and it does look as though someone has created an <a href="https://github.com/features/actions">Action</a> to fill this gap.</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/marketplace/actions/fork-sync-with-upstream" target="_blank" rel="noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:400/400;"><img src="https://avatars.githubusercontent.com/u/1312251?s=400&amp;amp;v=4" class="thumbnail" width="400" height="400"></div>

<h3><a href="https://github.com/marketplace/actions/fork-sync-with-upstream" target="_blank" rel="noopener">Fork Sync With Upstream - GitHub Marketplace</a></h3>


  <p><span class="label1">Automatically updates fork with new commits from upstream repo, safest to run on an unmodified branch</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Please note - this is not a GitHub product and we can’t provide direct support for it. I did try it out and it does work the way they describe - keeping the fork up to date with the upstream repository.</p>
<p>There are limitations to this - I’d strongly recommend having a careful read through the readme.</p>

**Comment by canuckjacq** [comment, marked as answer]
<p><a class="mention" href="/r-pankevicius">@r-pankevicius</a></p>
<p>I haven’t forgotten about you - we just shipped this feature:</p>
<aside class="onebox twitterstatus">
  <header class="source">
      <a href="https://twitter.com/github/status/1390382527588798477" target="_blank" rel="noopener">twitter.com</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/11161530/181065893-22032c9b-0fb4-4b65-a0b3-e8af9be8399d.png" class="thumbnail onebox-avatar" width="400" height="400">

<h4><a href="https://twitter.com/github/status/1390382527588798477" target="_blank" rel="noopener">GitHub (github)</a></h4>

<div class="tweet"> Is your fork behind? You can now sync from the parent repo with just a single click! 🖱💥 https://t.co/4Zx6OwVTxO</div>

<div class="date">
  <a href="https://twitter.com/github/status/1390382527588798477" target="_blank" rel="noopener">12:06 PM - 6 May 2021</a>
    <span class="like">
      <svg viewbox="0 0 512 512" width="14px" height="16px" aria-hidden="true">
        <path d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"></path>
      </svg> 7.9K
    </span>
    <span class="retweet">
      <svg viewbox="0 0 640 512" width="14px" height="16px" aria-hidden="true">
        <path d="M629.657 343.598L528.971 444.284c-9.373 9.372-24.568 9.372-33.941 0L394.343 343.598c-9.373-9.373-9.373-24.569 0-33.941l10.823-10.823c9.562-9.562 25.133-9.34 34.419.492L480 342.118V160H292.451a24.005 24.005 0 0 1-16.971-7.029l-16-16C244.361 121.851 255.069 96 276.451 96H520c13.255 0 24 10.745 24 24v222.118l40.416-42.792c9.285-9.831 24.856-10.054 34.419-.492l10.823 10.823c9.372 9.372 9.372 24.569-.001 33.941zm-265.138 15.431A23.999 23.999 0 0 0 347.548 352H160V169.881l40.416 42.792c9.286 9.831 24.856 10.054 34.419.491l10.822-10.822c9.373-9.373 9.373-24.569 0-33.941L144.971 67.716c-9.373-9.373-24.569-9.373-33.941 0L10.343 168.402c-9.373 9.373-9.373 24.569 0 33.941l10.822 10.822c9.562 9.562 25.133 9.34 34.419-.491L96 169.881V392c0 13.255 10.745 24 24 24h243.549c21.382 0 32.09-25.851 16.971-40.971l-16.001-16z"></path>
      </svg> 1.8K
    </span>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
